### PR TITLE
added data structure for backend models

### PR DESCRIPTION
**Pull request by rohit078689**
Added a raw structure and connections and login is yet to be build with other tasks

### CODE REVIEW

File: backend/models/Gametag.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package models
+
+import (
+	"github.com/google/uuid"
+)
+
+type GameTag struct {
```
Comment (jebidiah252) [inline backend/models/Gametag.go:7]: I think the game tags is something that is a little past MVP, but for now, I have no problem leaving this in here.

File: backend/models/UserRoles.go [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package models
+
+import (
+	"github.com/google/uuid"
+)
+
+type UserRoles struct {
```
Comment (jebidiah252) [inline backend/models/UserRoles.go:7]: I don't think that the roles should be a struct here. I was thinking that they could be an enum instead. I'd recommend taking a look at [this](https://www.sohamkamani.com/golang/enums/) article

File: backend/models/user.go [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package models
+
+import (
+	"github.com/google/uuid"
+)
+
+type User struct {
+	UserID       uuid.UUID
```
Comment (jebidiah252) [inline backend/models/user.go:8]: UserID is defined twice in this file.

**Review by jebidiah252 — CHANGES_REQUESTED** [review]
This looks like a great start here. One thing that I believe should be changed (and I'm looking at @ianjm24 while I say this) is that we could probably change all of the first IDs in each of our structs (i.e. in Game we have GameID) to just being ID (i.e. in the Game struct, we would replace GameID with ID). Let me know if that makes sense.

**Comment by ianjm24** [comment]
Yes I agree with @jebidiah252 about the IDs. The ID would be for the object itself, and a named ID would be for when one object references another (e.g. a Rating will have both an ID for itself and a GameID for the Game object it references)

File: backend/models/Gametag.go (hunk 1 above)
Comment (rohit078689) [reply backend/models/Gametag.go:7]: Implemented changes

**Comment by rohit078689** [comment]
Implemented suggested chanegs

**Review by jebidiah252 — APPROVED** [review]
Looks great!

File: backend/models/game.go [hunk 4]
```diff
@@ -5,12 +5,11 @@ import (
...
 type Game struct {
-	ID           uuid.UUID
-	Name         string
+	ID      uuid.UUID
+	Title        string
 	Description  string
 	UserID       uuid.UUID
-	CreationDate string
-	Version      string
+	VersionNumber string
```
Comment (ianjm24) [inline backend/models/game.go:12]: The game model will need to have a reference to where its game files are stored. For cloud storage this should just be a URL, so string should be fine.
```suggestion
	VersionNumber string
	GameStorageURL string
```

**Pull request merged**